clients/CMakeLists.txt

```diff
@@ -116,13 +116,40 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
 
   # if it fails to find OpenMP compile and link flags in strange configurations it can just use non-parallel reference computation
   # if there is no omp.h to find the client compilation will fail and this should be obvious, used to be REQUIRED
-  find_package(OpenMP)
+  if( CMAKE_CXX_COMPILER MATCHES ".*/hipcc$" )
+    find_package(OpenMP)
+    if (TARGET OpenMP::OpenMP_CXX)
+      set( LIBOMP_FOUND OpenMP::OpenMP_CXX)
+      get_filename_component(LIBOMP_PATH "${OpenMP_omp_LIBRARY}" PATH)
+    endif()
+  # Support for using gnu as cmake compiler.
+  else()
+    # libomp.so may be in a target subdirectory directory.
+    execute_process(
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      COMMAND bash -c "${HIP_CLANG_ROOT}/bin/amdclang --version | grep -oP '(?<=Target: ).+'"
+      OUTPUT_VARIABLE LLVM_TARGET_TRIPLE
+    )
+    find_library(LIBOMP_FOUND omp PATHS ${HIP_CLANG_ROOT}/lib PATH_SUFFIXES ${LLVM_TARGET_TRIPLE} NO_DEFAULT_PATH)
+    if (LIBOMP_FOUND)
+      get_filename_component(LIBOMP_PATH "${LIBOMP_FOUND}" PATH)
+    endif()
+  endif()
 
-  if (TARGET OpenMP::OpenMP_CXX)
-    set( COMMON_LINK_LIBS "OpenMP::OpenMP_CXX")
```

Comment:
So aren't we missing the OpenMP::OpenMP_CXX link request on CUDA backend?

Reply:
True, I can move `set( COMMON_LINK_LIBS ${LIBOMP_FOUND})` outside of the amd guard.